test/apple_test.dart

```diff
@@ -0,0 +1,209 @@
+    const buttonText = 'Sign up with Apple';
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: AppleSignInButton(
+            text: buttonText,
+          ),
+        ),
+      ),
+    );
+
+    expect(find.text(buttonText), findsOneWidget);
+  });
+
+  testWidgets('Check the default text style of the button is used',
+      (WidgetTester tester) async {
+    const defaultTextStyle = TextStyle(
+      fontSize: 16.0,
+      fontWeight: FontWeight.w500,
+      color: Colors.black,
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: AppleSignInButton(),
+        ),
+      ),
+    );
+
+    var text = find.byType(Text).evaluate().toList()[0].widget as Text;
+    expect(text.style.fontSize, defaultTextStyle.fontSize);
+    expect(text.style.fontWeight, defaultTextStyle.fontWeight);
+    expect(text.style.color, defaultTextStyle.color);
+  });
+
+  testWidgets('Check the provided text style of the button is used',
+      (WidgetTester tester) async {
+    const providedTextStyle = TextStyle(
+      fontSize: 22.0,
+      fontWeight: FontWeight.w900,
+      color: Colors.white,
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: AppleSignInButton(
+            textStyle: providedTextStyle,
+          ),
+        ),
+      ),
+    );
+
+    final AppleSignInButton microsoftSignInButton =
+        tester.firstWidget(find.byType(AppleSignInButton));
+    expect(microsoftSignInButton.textStyle, providedTextStyle);
+  });
+
+  testWidgets('Check supplied splash color is used',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: AppleSignInButton(
+            onPressed: () {},
+            splashColor: Colors.white,
+          ),
+        ),
+      ),
+    );
+
+    var button = find.byType(StretchableButton).evaluate().toList()[0].widget
+        as StretchableButton;
+    expect(button.splashColor, Colors.white);
+  });
+
+  testWidgets('Check default splash color is used',
+      (WidgetTester tester) async {
+    ButtonThemeData buttonTheme;
+
+    await tester.pumpWidget(MaterialApp(
+      home: Builder(
+        builder: (BuildContext context) {
+          buttonTheme = ButtonTheme.of(context);
+          return AppleSignInButton(
+            onPressed: () {},
+          );
+        },
+      ),
+    ));
+
+    var button =
+        find.byType(RaisedButton).evaluate().toList()[0].widget as RaisedButton;
+    expect(buttonTheme.getSplashColor(button),
+        buttonTheme.getTextColor(button).withOpacity(0.12));
+  });
+
+  testWidgets('Check the default AppleButtonStyle option is used',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: AppleSignInButton(),
+        ),
+      ),
+    );
+    final AppleSignInButton microsoftSignInButton =
```

Comment:
In a few places this file uses `microsoftSignInButton`, when it should be `appleSignInButton` or just `button`.

Reply:
Hi @dmjones, thanks for the review! I made necessary changes in my recent commits. Let me know what you think.

Reply:
Thanks!